lessons/ro/chapter_7.yaml

```
@@ -0,0 +1,272 @@
+- title: Capitolul 7 - Programare Orientata pe Obiecte
+  content_markdown: >
+    Exprimarea ideilor cu ajutorul functiilor este o metoda dovedita de a reprezenta
+    comportamente si date (Limbajul C o face de decenii!). 
+    
+    Din punct de vedere istoric, stiinta calculatoarelor a gasit alte agregatii si abstractii
+    expresive folositoare pentru date.
+    
+    Ati putea fi familiarizati cu programarea orientata pe obiecte (POO) ca fiind
+    una dintre acestea. In acest capitol vom explora limbajul Rust dincolo de functii.
+- title: Ce este POO?
+  content_markdown: >
+    Programarea orientata pe obiecte se refera, in linii mari, la limbajele de programare ce 
+    prezinta anumite caracteristici reprezentative:
+
+    * Incapsularea - Asocierea datelor si functiilor intr-o unitate conceptuala de un singur tip
+    numita *obiect*.
+
+    * Abstractizarea - Ascunderea datelor si functiilor pentru a acoperi detaliile 
+    implementarii unui obiect
+
+    * Polimorfismul - Abilitatea de a interactiona cu obiecte de tipuri diferite prin intermediul unei interfete.
+
+    * Mostenirea - Abilitatea de a mosteni date si comportamente de la alte obiecte.
+- title: Rust Nu Este POO
+  content_markdown: >
+    Rust nu dispune de o metoda semnificativa de mostenire a datelor si comportamentelor.
+
+    * Structurile nu pot mosteni campuri de la o structura parinte.
+
+    * Structurile nu pot mosteni functii de la o structura parinte.
+
+    Acestea fiind spuse, Rust implementeaza o multime de caracteristici ale unui limbaj de programare,
+    deci probabail nu va vor deranja aceste lipsuri.
+- title: Incapsulare prin matode
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=cfca759d42182509a46f65236bdfbf61
```

Use the [embedded code link](https://play.rust-lang.org/help#features-linking)

```suggestion
    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=struct%20CreaturaMaritima%20%7B%0A%20%20%20%20zgomot%3A%20String%2C%0A%7D%0A%0Aimpl%20CreaturaMaritima%20%7B%0A%20%20%20%20fn%20luare_zgomot(%26self)%20-%3E%20%26str%20%7B%0A%20%20%20%20%20%20%20%20%26self.zgomot%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20creatura%20%3D%20CreaturaMaritima%20%7B%0A%20%20%20%20%20%20%20%20zgomot%3A%20String%3A%3Afrom(%22blub%22)%2C%0A%20%20%20%20%7D%3B%0A%20%20%20%20println!(%22%7B%7D%22%2C%20creatura.luare_zgomot())%3B%0A%7D%0A
```